src/BenchmarkDotNet/Toolchains/DotNetCli/DotNetCliCommandExecutor.cs

```diff
@@ -23,13 +23,20 @@ public static class DotNetCliCommandExecutor
         public static DotNetCliCommandResult Execute(DotNetCliCommand parameters)
         {
             using (var process = new Process { StartInfo = BuildStartInfo(parameters.CliPath, parameters.GenerateResult.ArtifactsPaths.BuildArtifactsDirectoryPath, parameters.Arguments, parameters.EnvironmentVariables) })
-            using (var outputReader = new AsyncProcessOutputReader(process))
+            using (var outputReader = new AsyncProcessOutputReader(process, parameters.LogOutput, parameters.Logger))
             using (new ConsoleExitHandler(process, parameters.Logger))
             {
                 parameters.Logger.WriteLineInfo($"// start {parameters.CliPath ?? "dotnet"} {parameters.Arguments} in {parameters.GenerateResult.ArtifactsPaths.BuildArtifactsDirectoryPath}");
 
                 var stopwatch = Stopwatch.StartNew();
 
+                if (parameters.LogOutput && process.StartInfo.EnvironmentVariables.Keys.Count > 0)
```

Comment:
I like the idea of printing env vars, but I think that we should do it only once. With the current proposal we are going to print it for dotnet restore, build and publish.

I think that this logic could be moved to a private method of `DotNetCliCommand` class and invoked at the beginning of:

https://github.com/dotnet/BenchmarkDotNet/blob/937311cb62ac5f3ffab69e243e502d4bc20cd015/src/BenchmarkDotNet/Toolchains/DotNetCli/DotNetCliCommand.cs#L47-L48

https://github.com/dotnet/BenchmarkDotNet/blob/937311cb62ac5f3ffab69e243e502d4bc20cd015/src/BenchmarkDotNet/Toolchains/DotNetCli/DotNetCliCommand.cs#L71-L72